tests/cpp_caching/YSettings

```diff
@@ -0,0 +1,23 @@
+def known_flavors():
+    return ['debug', 'release']
+
+def default_flavor():
+    return 'release'
+
+def get_common_config(config, args):
+    clang = 'clang++-8'
+    return {
+        'compiler': clang,
+        'linker': clang,
+        'compile_flags': ['-std=c++14', '-Wall', '-fcolor-diagnostics'],
+    }
+
+def get_flavored_config(config, args):
+    return {
+        'release': {
+            'compile_flags': ['$*', '-O2', '-DDEBUG'],
+        },
+        'debug': {
+            'compile_flags': ['$*', '-g', '-DDEBUG'],
+        },
+    }.get(config.flavor, {})
```

Comment:
is this all needed for the test? wouldn't it work the same with default values for everything?

Reply:
I will try simplifying that

Reply:
I looked at it with @dlehavi-resonai and we didn't find too many things that can be removed